source/common/stats/stats_matcher_impl.cc

```diff
@@ -0,0 +1,44 @@
+#include "common/stats/stats_matcher_impl.h"
+
+#include <regex>
+#include <string>
+
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Stats {
+
+StatsMatcherImpl::StatsMatcherImpl(const envoy::config::metrics::v2::StatsConfig& config) {
+  switch (config.stats_matcher().stats_matcher_case()) {
+  case envoy::config::metrics::v2::StatsMatcher::kExclusionList:
+    for (const auto& stats_matcher : config.stats_matcher().exclusion_list().patterns()) {
+      matchers_.push_back(Matchers::StringMatcher(stats_matcher));
+    }
+    break;
+  case envoy::config::metrics::v2::StatsMatcher::kInclusionList:
+    default_inclusive_ = false;
```

Comment:
nit: I think it would be a bit clearer if this was only initialized in this function (rather than relying on the default that's defined in the header for some cases). To do this without typing `default_inclusive_ = true` twice (if that's the reason it's annoying to initialize it here), you should be able to use a fallthrough to partially combine the `kExclusionList` and `default` cases.

Reply:
Done.